expressions.go

```diff
@@ -319,6 +319,36 @@ var functions = map[string]struct {
 			return nil, fmt.Errorf("sum expects argument #1 to be slice of numbers")
 		}
 	}},
+	"sumiftrue": {2, func(args []Value) (Value, error) {
+		arg := args[0]
```

Comment:
Easier to write this part as

```
values, ok := args[0].(*Slice)
if !ok || values.elementype <> number type { ... expect arg 1 to be number slice }

conditions, ok := args[1].(*Slice)
if !ok || conditions.elementype <> bool type {  .. expect arg 2 to be bool slice  }

if len(values) != len(conditions) { ... expect both slices of same size }

// then proceed to do the actual calc
```
And spacing here is important: 1. validate arg1, 2. validate arg3, 3. validate well formdness of call based on dynamic data (in the sense that 1 & 2 could be done statically if we had a static type system).

Reply:
Your comment about handing `undefined` applies here too, so would need to do something a tad more verbose so return `undefined` is arg0 or arg1 are `undefined` themselves.

Reply:
I don't think we need that here. A selected slice will have the right element type, even if the slice is empty or the elements are undefined.